### ISSUE TITLE
Testing

### ISSUE THREAD

**Issue by majormoses** [sensu/sensu-go-chef#6]
Before we start accepting work we need to agree on everything needs to be tested with real integration tests. I think this is what we should have as the bare minimum:
- all PRs are should run tests

We should use the following tools:
- `travis-ci`
- `test-kitchen`
- `kitchen-do(kken|cker)`
- `(server|in)spec` I am torn on this I think we should use `serverspec` as it will be easier for other CM projects (puppet, salt, ansible, etc) to take what we have and drop it in place. Also the plugins currently use `serverspec` but I am open to hearing counter arguments for inspec.

**Comment by majormoses** [comment]
I have enabled travis to test if a `.travis.yml` is present.

**Comment by webframp** [comment]
I don't have a strong preference regarding `(server|in)spec` since they are similar enough but here's a few observations.

1. In general `inspec` seems to be much faster at executing verification steps.
1. `inspec` profiles can be stored in a separate shared repo. This is a nice feature if you are considering shared test profiles for other CM tools. Example:
   * Inspec Profile: https://github.com/dev-sec/linux-baseline
   * Cookbook usage: https://github.com/dev-sec/chef-os-hardening/blob/master/test/integration/default/inspec.yml
1. Partner guidelines seem to suggest inspec: https://github.com/chef-partners/cookbook-guide/blob/master/docs/12_integration_tests.md

**Comment by majormoses** [comment]
> profiles can be stored in a separate shared repo

Hm that would mean that you would need two PRs then for every contribution then as you would need to update in two repos to test.

> Partner guidelines seem to suggest inspec

I will reach out to someone at chef to clarify if that is a requirement or a we'd like you to use our stuff.

**Comment by eheydrick** [comment]
+1 for inspec. Also I think we should test on Ubuntu 18.04 as that's just about out.

**Comment by webframp** [comment]
@jjasghar you're amazing. Thanks for sharing that awesome example! I also lean towards dokken.

**Comment by webframp** [comment]
@eheydrick I agree. It doesn't look like any [packages for ubuntu/bionic](https://packagecloud.io/app/sensu/nightly/search?q=&filter=debs&filter=debs&dist=ubuntu%2Fbionic) are pushed to the repos yet but when they are it can be added as a testing platform.

**Comment by jaredledvina** [comment]
I'm been working on using Kitchen for the Ansible role as well. So far, using `kitchen-docker` has been a nightmare to get working w/ systemd and init systems in general. Was thinking about trying out `kitchen-vagrant` instead. I'm totally happy to align testing strategies across the CM repo's. I've also started with inspec and it's proving out well. 
I'm curious if folks have been able to get a setup with `kitchen-docker` to work across `debian`, `centos`, `ubuntu` when it comes to testing out multiple services inside the container.

**Comment by jaredledvina** [comment]
Ahhhh nice! I'm going to steal your `.kitchen.yml` from there and give it a whirl, I was using `kitchen-docker` and most likely over complicating the setup. Once I have something complete for the v0 of the Ansible work, I'll get it opened up (totally mess currently). I also hang out on the Community Slack server, feel free to reach out whenever!

**Comment by webframp** [comment]
Testing is overhauled also in #7 but there's an open question about rubocop config for chef partner cookbooks. Could benefit from the expertise of @jjasghar .

**Comment by webframp** [comment]
@majormoses Are the recently merged changes sufficient to close this?
